vmdb/app/controllers/provider_foreman_controller.rb

```diff
@@ -454,25 +470,35 @@ def provider_node(id, model)
   end
 
   def configuration_profile_node(id, model)
-    cpf = identify_record(id, ConfigurationProfile)
-    if cpf.nil?
+    @record = @configuration_profile_record = identify_record(id, ConfigurationProfile)
+    if @configuration_profile_record.nil?
       self.x_node = "root"
       get_node_info("root")
       return
     else
       options = {:model => "ConfiguredSystem"}
-      options[:where_clause] = ["configuration_profile_id IN (?)", cpf.id]
+      options[:where_clause] = ["configuration_profile_id IN (?)", @configuration_profile_record.id]
       process_show_list(options)
       record_model = ui_lookup(:model => model ? model : TreeBuilder.get_model_for_prefix(@nodetype))
-      @right_cell_text =
+      if @sb[:active_tab] == 'configured_systems'
+        @right_cell_text =
           _("%{model} \"%{name}\"") %
-          {:name  => cpf.name,
+          {:name  => @configuration_profile_record.name,
            :model => "#{ui_lookup(:tables => "configured_system")} under #{record_model}"}
+      else
+        @showtype = 'main'
+        @pages = nil
+        @right_cell_text =
+          _("%{model} \"%{name}\"") %
+          {:name  => @configuration_profile_record.name,
+           :model => record_model
+          }
+      end
     end
   end
 
   def configured_system_node(id, model)
-    @record = identify_record(id, ConfiguredSystem) unless @record
+    @record = @configured_system_record = identify_record(id, ConfiguredSystem)
```

Comment:
did you want?
```ruby
@record ||= @configured_system_record = identify_record(id, ConfiguredSystem)
```

Or did you want to change the original conditional assignment?